src/aiida/manage/configuration/settings.py

```diff
@@ -66,45 +66,94 @@ def create_instance_directories() -> None:
+        * If an existing directory is still not found, the ``DEFAULT_AIIDA_PATH`` is used.
 
-    In principle then, a configuration folder should always be found or automatically created.
+    :returns: The path of the configuration directory.
     """
-    global AIIDA_CONFIG_FOLDER  # noqa: PLW0603
-    global DAEMON_DIR  # noqa: PLW0603
-    global DAEMON_LOG_DIR  # noqa: PLW0603
-    global ACCESS_CONTROL_DIR  # noqa: PLW0603
+    dirpath_config: pathlib.Path | None = None
+
+    if environment_variable := os.environ.get(DEFAULT_AIIDA_PATH_VARIABLE):
+        dirpath_config = get_configuration_directory_from_envvar(environment_variable)
+    else:
+        dirpath_config = get_configuration_directory_from_cwd()
 
-    if aiida_config_folder is not None:
-        AIIDA_CONFIG_FOLDER = aiida_config_folder
+    # If no existing configuration directory is found, fall back to the default
+    if dirpath_config is None:
+        dirpath_config = pathlib.Path(DEFAULT_AIIDA_PATH).expanduser() / DEFAULT_CONFIG_DIR_NAME
 
-    elif environment_variable := os.environ.get(DEFAULT_AIIDA_PATH_VARIABLE):
-        # Loop over all the paths in the `AIIDA_PATH` variable to see if any of them contain a configuration folder
-        for base_dir_path in [path for path in environment_variable.split(':') if path]:
-            AIIDA_CONFIG_FOLDER = pathlib.Path(base_dir_path).expanduser()
+    return dirpath_config
 
-            # Only add the base config directory name to the base path if it does not already do so
-            # Someone might already include it in the environment variable. e.g.: AIIDA_PATH=/home/some/path/.aiida
-            if AIIDA_CONFIG_FOLDER.name != DEFAULT_CONFIG_DIR_NAME:
-                AIIDA_CONFIG_FOLDER = AIIDA_CONFIG_FOLDER / DEFAULT_CONFIG_DIR_NAME
 
-            # If the directory exists, we leave it set and break the loop
-            if AIIDA_CONFIG_FOLDER.is_dir():
-                break
-    else:
-        # The `AIIDA_PATH` variable is not set so use the default path and try to create it if it does not exist
-        AIIDA_CONFIG_FOLDER = pathlib.Path(DEFAULT_AIIDA_PATH).expanduser() / DEFAULT_CONFIG_DIR_NAME
+def get_configuration_directory_from_envvar(environment_variable: str) -> pathlib.Path:
+    """Return the path of a config directory from.
+
+    The environment variable should be a colon separated string of filepaths that either point directly to a config
+    directory or a path that contains a config directory. The first match is returned. If no existing config directory
+    is found, the last path in the environment variable is used.
+
+    :returns: The path of the configuration directory.
+    """
+    # Loop over all the paths in the ``AIIDA_PATH`` variable to see if any of them contain a configuration folder
+    for base_dir_path in [path for path in environment_variable.split(':') if path]:
+        dirpath_config = pathlib.Path(base_dir_path).expanduser()
```

Comment:
What was the exact way in which this broke for you? I cannot reproduce it. For example:
```
In [2]: get_configuration_directory_from_envvar('.:/tmp')
Out[2]: PosixPath('/tmp/.aiida')
```
if there is no `.aiida` folder in the CWD. If there _is_ such a folder, that path is used. So it seems to me like it is working as intended.

That being said, I notice that there is no validation whatsoever for the value in `AIIDA_PATH`. I wonder if, rather than trying to accommodate for all possible weird cases like `.`, we should really be validating and raising for invalid `AIIDA_PATH` values.
Validation could be:
1. Should be a string
2. When split on `:`, all elements should be _absolute_ filepaths (allowing relative paths is for sure going to cause issues)
3. Each path (omitting the `.aiida` leaf if present) _has_ to exist. If it doesn't exist, it is ignored.

Now this may be backwards incompatible, but I think if someone is using anything that does not respect these rules, chances are high that things can easily break anyway.

Reply:
My issue was prior to this PR. I tested it with and without an empty `.aiida` directory present (as that was the use case), both in the first and second directory that was set via `AIIDA_PATH`, and using the `verdi profile setup core.sqlite_zip` command. This was part of the discussion on what is the fastest way to quickly explore an aiida-archive without polluting the main profile (before your actual `verdi init` PR). In both cases, nested `.aiida/.aiida` trees were generated, and, consequently, the exception:
`FileNotFoundError: [Errno 2] No such file or directory: '.aiida/access/<profile>'`
was thrown as the actual location became `'.aiida/.aiida/access/<profile>'`.

I just tried the same with the updated code, but still getting the same issue, while the pure Python function gives the same output as what you posted. I suppose there might be more things happening when runnnig `verdi profile setup core.sqlite_zip`?

However, I am also strongly in favor of doing some validation on `AIIDA_PATH`. Is it OK if I create a commit directly on this PR?

Reply:
> However, I am also strongly in favor of doing some validation on AIIDA_PATH. Is it OK if I create a commit directly on this PR?

Since it seems to be an unrelated issue to the changes of this PR, I think it would be best to reserve that for a separate PR. Especially since adding validation would potentially be backwards incompatible, it would be best to not mix this here. I think we also need to discuss the coming team meeting if others agree to add the proposed backwards incompatibility.

Reply:
Pasting the testing I did prior to this PR here for completeness (can later be moved to a separate PR):

---

Some notes:
- `AIIDA_PATH` was set to `'.'` from the parent folder in which the `'.aiida'` directory was supposed to be created (not that it should actually make a difference), and the `verdi profile setup core.sqlite_zip` was run from this folder
- Manually added `'.aiida'` folders were always empty; apart from that I only had my global `'~/.aiida'`
- Warning = UserWarning: Creating AiiDA configuration folder `'.aiida'`
- :x: = Breaks with `FileNotFoundError` due to nested `'.aiida/.aiida'`

---

1. AIIDA_PATH='.'

|                                | No second path                                       |
| ------------------------------ | ---------------------------------------------------- |
| First path -> '.aiida' present | :x: '.aiida/.aiida' tree in first path               |
| First path -> '.aiida' absent  | :x: Warning, then '.aiida/.aiida' tree in first path |

---

2. AIIDA_PATH='<first_path>'

|                                | No second path                                                  |
| ------------------------------ | --------------------------------------------------------------- |
| First path -> '.aiida' present | :white_check_mark: '.aiida' created in first path               |
| First path -> '.aiida' absent  | :white_check_mark: Warning, then '.aiida' created in first path |

---

3. AIIDA_PATH='.:<second_path>'

|                                | Second path -> '.aiida' present                    | Second path -> '.aiida' absent                                   |
| ------------------------------ | -------------------------------------------------- | ---------------------------------------------------------------- |
| First path -> '.aiida' present | :x: '.aiida/.aiida' tree in first path             | :x: '.aiida/.aiida' tree in first path                           |
| First path -> '.aiida' absent  | :white_check_mark: '.aiida' created in second path | :white_check_mark: Warning, then '.aiida' created in second path |

---

4. AIIDA_PATH='<first_path>:<second_path>'

|                                | Second path -> '.aiida' present                      | Second path -> '.aiida' absent                                     |
| ------------------------------ | ---------------------------------------------------- | ------------------------------------------------------------------ |
| First path -> '.aiida' present | :white_check_mark: '.aiida' created in first path    | :white_check_mark: '.aiida' created in first path                  |
| First path -> '.aiida' absent  | :white_check_mark: '.aiida' created in _second_ path | :white_check_mark: Warning, then '.aiida' created in _second_ path |

---

So yeah, without `'.'` things work as expected, but with it, hell breaks loose. You're right ,better to keep it separate and discuss it with the rest of the team :rocket:

Reply:
Thanks for the detailed analysis @GeigerJ2 . I think this shows that validation of `AIIDA_PATH` is in order 😅   Is there anything else for this PR remaining, or can we merge this?

Reply:
From my side, it's fine! Would have approved the merge directly, but not sure if I am ready to wield such power yet :D Thanks for the changes, I think they go in a good direction!